crates/modelardb_server/src/storage/uncompressed_data_manager.rs

```diff
@@ -93,61 +117,110 @@ impl UncompressedDataManager {
                 .map_err(|error| IOError::new(IOErrorKind::InvalidData, error))?;
 
             // unwrap() is safe as univariate_id can only exist if it is in the metadata database.
-            let error_bound = metadata_manager.error_bound(univariate_id).await.unwrap();
+            let table_name = context
+                .metadata_manager
+                .univariate_id_to_table_name(univariate_id)
+                .await
+                .unwrap();
+
+            // unwrap() is safe as data cannot be ingested into a model table that does not exist.
+            let model_table_metadata = context
+                .model_table_metadata_from_default_database_schema(&table_name)
+                .await
+                .unwrap()
+                .unwrap();
 
             for maybe_file_dir_entry in folder_dir_entry.path().read_dir()? {
-                finished_data_buffers.push_back(Box::new(UncompressedOnDiskDataBuffer::try_new(
+                let buffer = Box::new(UncompressedOnDiskDataBuffer::try_new(
                     univariate_id,
-                    error_bound,
+                    model_table_metadata.clone(),
                     maybe_file_dir_entry?.path(),
-                )?));
+                )?);
+
+                initial_disk_space += buffer.disk_size();
+
+                self.channels
+                    .univariate_data_sender
+                    .send(Message::Data(buffer))
```

Comment:
The previously spilled buffers are immediately send to the compressor as it has nothing to do until enough new data points are pushed to the system to create new full buffers, so we may as well spend the time to compress the previously spilled buffers.

Reply:
I agree that it is fine to do but it can be a bit hard to spot what exactly happens when we send messages across the channels and the `initialize` function does not mention we compress directly. Maybe change the doc comment or add a comment here mentioning that we are compressing the previously spilled data immediately.

Reply:
Ok, I have added a doc comment.